### ISSUE TITLE
Add Directive Comment

### ISSUE THREAD

**Issue by zero-plusplus** [zero-plusplus/vscode-autohotkey-debug#30]
Like [debugger](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/debugger) statement of JavaScript.

This extension recognizes comments following the rules as embedded breakpoints.
To function as an embedded breakpoint, the following must be observed.
1. it must be a single-line comment
2. only a space is allowed before the comment
3. the comment is followed by an optional space and a `debugger`. e.g. `; debugger`

The need to parse the source code reduces the performance at startup. Therefore, you need to set `useEmbeddedBreakpoint` to true in launch.json to use this feature.

Also, condition and log point can be set just like the actual breakpoints.
The rules for this have not been decided yet, but here is a draft.

1. Separate each one with a colon. Each is optional.
  1-1. e.g. `debugger:CONDITION:HITCONDITION:LOGPOINT`
  1-2. e.g. `debugger:CONDITION::LOGPOINT`
  1-3. e.g. `debugger::HITCONDITION:LOGPOINT`
  1-4. e.g. `debugger:::LOGPOINT`
2. enclose in parentheses
  2-1. `debugger(CONDITION)[HITCONDITION]{LOGPOINT}`
  2-2. `debugger(CONDITION){LOGPOINT}`
  2-3. `debugger[HITCONDITION]{LOGPOINT}`
  2-4. `debugger{LOGPOINT}`

`1.` is simple and good. The conditions are easy to parse because they do not use colons. However, it may be difficult to tell at a glance which role it plays when it is omitted.

In the case of `2.`, unlike `1.`, the role is easy to understand even if it is omitted. It would be easy to parse. However, it is very unattractive.

If you have any suggestions, please comment.

**Comment by zero-plusplus** [comment]
The `useEmbeddedBreakpoint` changes to a property of `useAdvancedBreakpoint`.

That is, it looks like the following.
```
...
"useAdvancedBreakpoint": {
    "useEmbeddedBreakpoint": true
}
...
```

**Comment by zero-plusplus** [comment]
I would review the idea of adding a `useEmbeddedBreakpoint` property to `useAdvancedBreakpoint` because an embedded breakpoint makes use of three things.

This is because an Embedded breakpoint makes use of three things
1. Conditional breakpoint
2. Hit Conditional breakpoint
3. Log point

This means that `useAdvancedBreakpoint` must be enabled as a prerequisite.
The addition of the `useEmbeddedBreakpoint` property makes it look as if the above three can be set separately.

This will mislead many people.

In conclusion, it would be better to set them as completely separate settings, such as `useExtraBreakpoint`.

#31 is the same.

**Comment by zero-plusplus** [comment]
I noticed that it doesn't matter whether Advanced breakpoint is enabled or not, because Embedded breakpoint simply takes the place of setting breakpoints.

I can simply add `useEmbeddedBreakpoint`.

**Comment by zero-plusplus** [comment]
I'm going to add some other features.

A draft is below.

## Embedded Logging
This is a specialized feature for log output.
```ahk
; log:LEVEL -> MESSAGE
; log:LEVEL => MESSAGE WITH A NEW LINE

; log:LEVEL ->| NOT LEADING SPACES
; log:LEVEL =>| NOT LEADING SPACES WITH A NEW LINE
```

Here are some of the advantages over embedded breakpoints.
1. The output category can be changed by log level. vscode can be color-coded by output category
2. The log output can be selected according to the level.

### Settings
Output all categories
```json
"useEmbeddedLogging": true
```

Select individual logs to be output by level.
```json
"useEmbeddedLogging": {
    "info": true,
    "error": true
}
```

Shorthand above.
```json
"useEmbeddedLogging": [ "info", "error" ]
```

**Comment by SAbboushi** [comment]
```
x := 3
; debug
```
If `useEmbeddedBreakpoint` is enabled, then execution breaks at debug comment.  Benefit is that I set breakpoint in script instead of in vscode UI?

Maybe I am too late to provide feedback - it took me awhile to understand the feature you are working on.
I prefer "2) enclose in parentheses"

**Comment by zero-plusplus** [comment]
This feature has been created as an alternative to `Advanced Output`.
In other words, it is intended to be used as a log output.

I've never used JavaScript's debugger statements, so probably no one will use them as a breakpoint.

After thinking about it, I don't seem to need an `Embedded breakpoint` if I have `Embedded Logging`.

`Embedded Logging` can't control whether or not it outputs according to the conditions, so it seems to be good if only that is improved.

**Comment by SAbboushi** [comment]
oops - sorry I closed this!

**Comment by zero-plusplus** [comment]
It's okay. I was the one who closed it.
I opened it up again as the functionality was extended.

**Comment by zero-plusplus** [comment]
I'm worried about the performance degradation because it searches the entire source code.
But I tried with 7000 lines of source code and it took about 0.2 seconds, the second time was 0.07 seconds thanks to the cache.

Since this is an Intel(R) Core i7 9700k CPU result, it may take about 1 second on a laptop.

**Comment by SAbboushi** [comment]
> I just realized that you sponsored me. Thank you!
You are welcome.  I plan to for 1 year.  Very small contribution... Wish I could help you more.

> Since this is an Intel(R) Core i7 9700k CPU result, it may take about 1 second on a laptop.
I think that is very reasonable tradeoff - especially since the default option for this feature is disabled (i.e. performance degredation only for those of us who intentionally enable).

**Comment by zero-plusplus** [comment]
I'm happy enough!

I'm more than happy to be recognized for what I've created.

**Comment by zero-plusplus** [comment]
Each feature in the draft will be a separate issues.
